service/dap/server.go

```diff
@@ -1859,6 +1876,53 @@ func (s *Server) childrenToDAPVariables(v *fullyQualifiedVariable) ([]dap.Variab
 				Value:              cvalue,
 				VariablesReference: cvarref,
 				IndexedVariables:   getIndexedVariableCount(c),
+				NamedVariables:     getNamedVariableCount(c),
+			}
+		}
+	}
+	return children, nil
+}
+
+func getNamedVariableCount(v *proc.Variable) int {
+	namedVars := 0
+	if len(v.Children) > 0 && (v.Kind == reflect.Array || v.Kind == reflect.Slice) {
+		childKind := v.Children[0].RealType.Common().ReflectKind
+		if len(v.Children) > 0 && (childKind == reflect.Uint8 || childKind == reflect.Int32) {
+			// string value of array/slice of bytes and runes.
+			namedVars += 1
+		}
+	}
+	return namedVars
+}
+
+// toNamedChildren returns the DAP presentation of the referenced variable's named children.
+// Named variables for this server are used to send additional information about a variable
+// that is separate from the indexed variables.
+func (s *Server) toNamedChildren(v *fullyQualifiedVariable) ([]dap.Variable, error) {
+	var children []dap.Variable
+
+	if len(v.Children) > 0 && (v.Kind == reflect.Array || v.Kind == reflect.Slice) {
+		childKind := v.Children[0].RealType.Common().ReflectKind
+		if len(v.Children) > 0 && (childKind == reflect.Uint8 || childKind == reflect.Int32) {
+			// Return the string value of []byte or []rune.
+			typeName := api.PrettyTypeName(v.DwarfType)
+			loadExpr := fmt.Sprintf("string(*(*%q)(%#x))", typeName, v.Addr)
+
+			s.log.Debugf("loading %s (type %s) with %s", v.fullyQualifiedNameOrExpr, typeName, loadExpr)
+			// We know that this is an array/slice of Uint8 or Int32, so we will load the full array.
+			// TODO(suzmue): should we cap this at or near MaxStringLen?
+			config := DefaultLoadConfig
+			config.MaxArrayValues = int(v.Len)
```

Comment:
This is going to be trouble if for some reason the variable is uninitialized, I think you want `config.MaxArrayValues = config.MaxStringLen`

Reply:
Done.

Reply:
I think the issue was that v.Len may not be a value that makes sense, so the increase in limit should not be an issue.